pennylane/capture/dynamic_shapes.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2025 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains a utility for handling inputs with dynamically shaped arrays.
+"""
+from functools import lru_cache
+from string import ascii_lowercase as letters
+
+has_jax = True
+try:
+    import jax
+except ImportError:  # pragma: no cover
+    has_jax = False  # pragma: no cover
+
+
+@lru_cache
+def _get_letter(ind: int) -> str:
+    if ind < 26:
+        return letters[ind]
+    if ind < 702:
+        return letters[ind // 26 - 1] + letters[ind % 26]
+    raise NotImplementedError("we only support up to 702 dynamic axes")  # pragma: no cover
```

Comment:
```suggestion
def _get_letter(ind: int) -> str:
    out = ""
    while ind > 0:
        ind, r = divmod(ind, 26)
        out += letters[r]

    return out[::-1]
```

Reply:
Given it starts getting harder to test at higher indices too, I'm going to stick with the version I can nicely confirm continues working.